src/bun.js/api/glob.zig

```diff
@@ -252,14 +243,11 @@ fn makeGlobWalker(
     const only_files = matchOpts.only_files;
 
     if (cwd != null) {
-        var globWalker = alloc.create(GlobWalker) catch {
-            globalThis.throwOutOfMemory();
-            return error.JSError;
-        };
+        var globWalker = try alloc.create(GlobWalker);
```

Comment:
according to zack this allocation is not by an arena. i feel like theres alot of these patterns in the codebase, but we can now use errdefer to prevent memory leaks in error paths.